src/handlers/decision.rs

```diff
@@ -152,6 +162,119 @@ pub(super) async fn handle_command(
     }
 }
 
+fn build_status_comment(
+    history: &BTreeMap<String, Vec<UserStatus>>,
+    current: &BTreeMap<String, Option<UserStatus>>,
+) -> anyhow::Result<String> {
+    let mut comment = "| Team member | State |\n|-------------|-------|".to_owned();
+    for (user, statuses) in history {
+        let mut user_statuses = format!("\n| {} |", user);
+
+        // previous stasuses
+        for status in statuses {
+            let status_item = format!(" ~~{}~~ ", resolution_to_str(&status.resolution));
+            user_statuses.push_str(&status_item);
+        }
+
+        // current status
+        let current_status = current.get(user).unwrap();
+        let user_resolution;
+        match current_status {
+            Some(status) => user_resolution = resolution_to_str(&status.resolution),
+            _ => user_resolution = "".to_string(),
+        }
+        let status_item = format!(" **{}** |", user_resolution);
+        user_statuses.push_str(&status_item);
+
+        comment.push_str(&user_statuses);
+    }
+
+    println!("{}", comment);
+    Ok(comment)
+}
+
+fn resolution_to_str(resolution: &Resolution) -> String {
+    match resolution {
+        Merge => "merge".to_owned(),
+        Hold => "hold".to_owned(),
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_build_status_comment() {
+        let mut history: BTreeMap<String, Vec<UserStatus>> = BTreeMap::new();
+        let mut current_statuses: BTreeMap<String, Option<UserStatus>> = BTreeMap::new();
+
+        // user 1
+        let mut user_1_statuses: Vec<UserStatus> = Vec::new();
+
+        user_1_statuses.push(UserStatus {
+            comment_id: "some-id".to_string(),
+            text: "I like this".to_string(),
+            reversibility: Reversibility::Reversible,
+            resolution: Resolution::Merge,
+        });
+
+        user_1_statuses.push(UserStatus {
+            comment_id: "some-id".to_string(),
+            text: "I like this".to_string(),
+            reversibility: Reversibility::Reversible,
+            resolution: Resolution::Hold,
+        });
+
+        history.insert("Niklaus".to_string(), user_1_statuses);
+
+        current_statuses.insert(
+            "Niklaus".to_string(),
+            Some(UserStatus {
+                comment_id: "some-id".to_string(),
+                text: "I like this".to_string(),
+                reversibility: Reversibility::Reversible,
+                resolution: Resolution::Merge,
+            }),
+        );
+
+        // user 2
+        let mut user_2_statuses: Vec<UserStatus> = Vec::new();
+
+        user_2_statuses.push(UserStatus {
+            comment_id: "some-id".to_string(),
+            text: "I like this".to_string(),
+            reversibility: Reversibility::Reversible,
+            resolution: Resolution::Hold,
+        });
+
+        user_2_statuses.push(UserStatus {
```

Comment:
Might be a good idea to build something like a `UserStatus` factory? So we don't repeat it so many times here? Anyway, I'm not 100% sure of what I'm saying 😄

Reply:
💯!
I'll give it a try